Sources/NIO/EventLoop.swift

```diff
@@ -622,6 +626,9 @@ public protocol EventLoopGroup: class {
     /// The virtue of this function is to shut the event loop down. To work around that we call back on a DispatchQueue
     /// instead.
     func shutdownGracefully(queue: DispatchQueue, _ callback: @escaping (Error?) -> Void)
+
+    /// The internal `EventLoop`s of the group. This array can be used to iterate over every `EventLoop`.
```

Comment:
we should do two things here:

- document that it's optional because not all event loop groups support it
- file a new issue (tagged as 2.0) to make this mandatory & link the issue here

also I'd like to discuss if an array is really the right API here. Mandating an array might force an allocation if some event loop group doesn't store it's event loops in an array already. Maybe we should use an iterator. Unsure. Maybe we can leave it like this because for 2.0 we'll change this API anyway. @Lukasa / @normanmaurer opinions if array or iterator is better here? I'm leaning towards iterator

Reply:
When you say "use an iterator", what specific return type did you have in mind?

Reply:
If we talk about using an iterator like my alternative solution than an allocation is happening as well since it's passed by value right? Could you explain how one can avoid an allocation with an iterator?

Reply:
@Lukasa new type `struct EventLoopGroupIterator`

Reply:
@FranzBusch well the `EventLoopGroupIterator` could be constructed with different kinds of things. Say `[EventLoop]` if you already have one, or `IndexingIterator<Array<Element>>` or something. The idea is that we can hide what we're storing under the hood and just implement the `Iterator` protocol.